scene/gui/text_edit.cpp

```diff
@@ -1157,9 +1152,9 @@ void TextEdit::_notification(int p_what) {
 				}
 
 				if (rtl) {
-					RenderingServer::get_singleton()->canvas_item_add_rect(ci, Rect2(size.width - (xmargin_end + 2) - minimap_width, viewport_offset_y, minimap_width, viewport_height), viewport_color);
+					RenderingServer::get_singleton()->canvas_item_add_rect(ci, Rect2(size.width - xmargin_end - minimap_width, viewport_offset_y, minimap_width, viewport_height), viewport_color);
```

Comment:
I initially removed the extra 2 pixels added to the end margin while debugging this issue, since I suspected they might be the cause. I’ll revert that change, as they weren’t actually responsible.